### ISSUE TITLE
linear() doesn't accept formula as object

### ISSUE THREAD

**Issue by Cdevenish** [TheoreticalEcology/s-jSDM#61]
Doesn't seem possible to add formula as object to `sjDM()` with `linear()`
```
set.seed(42)
# simulate data
community <- simulate_SDM(sites = 100, species = 10, env = 3)
Env <- community$env_weights

Env <- as.data.frame(Env)

# make formula
form1 <- as.formula(~V1+V2+V3)
form1

Env.lin1 <- linear(data = Env, formula = form1) # this throws an error: (Error: object of type 'symbol' is not subsettable)

Env.lin2 <- linear(data = Env, formula = ~V1+V2+V3) # this is OK

```

**Comment by MaximilianPi** [comment]
Hi @Cdevenish,

You are correct, I have found the problem:

Currently, an expression is expected to be passed to the formula argument and any variable that is passed is treated as 'name' object and the name of the variable is passed to the internal as.formula function. 

The solution would be to catch non-expressions and evaluate them in the parent's environment. 

I will push a fix with the next subversion.

Many thanks.

**Comment by MaximilianPi** [comment]
Solved with version 1.6
